tools/automator.sh

```
@@ -0,0 +1,176 @@
+  echo -e "  -f <oauth-token-file>\t File that contains the GitHub oauth token. Mandatory."
+  echo -e "  -c <command>\t\t Command to execute to generate the diff to be committed. Mandatory."
+  echo -e "  -b <branch>\t\t Branch. Defaults to the same as the original repo."
+  echo -e "  -l <label>\t\t Labels (comma-separated) to add to PR."
+  echo -e "  -t <title>\t\t Title of PR. "
+
+  if [ -n "${1:-}" ]  ; then
+    echo -e "\n${1}"
+    exit 1
+  fi
+
+  exit 0
+}
+
+function parse_args() {
+
+  while getopts ":o:r:f:c:b:l:t:h" opt; do
+    case ${opt} in
+      o) ORG=$(trim "${OPTARG}");;
+      r) REPO=$(trim "${OPTARG}");;
+      f) TOKEN_FILE=$(trim "${OPTARG}");;
+      c) CMD=$(trim "${OPTARG}");;
+      b) BRANCH=$(trim "${OPTARG}");;
+      l) LABELS=$(trim "${OPTARG}");;
+      t) TITLE=$(trim "${OPTARG}");;
+      h) usage 0;;
+      *) usage "Invalid command line argument. Aborting." ;;
+    esac
+  done
+
+  shift $((OPTIND -1))
+  if [ $# -ne 0 ]; then
+    usage "Invalid command line argument. Aborting."
+  fi
+
+  validate_args
+}
+
+function validate_args() {
+  [ -z "${ORG:-}" ] && usage "Missing organization"
+  [ -z "${REPO:-}" ] && usage "Missing repository"
+  [ -z "${CMD:-}" ] && usage "Missing command"
+  [ -z "${TOKEN_FILE:-}" ] && usage "Missing token file"
+
+  TOKEN="$(cat "${TOKEN_FILE}")"
+  [ -z "${TOKEN}" ] && usage "Invalid token file"
+
+  BRANCH="${BRANCH:-$(git describe --contains --all HEAD)}"
+  TITLE="${TITLE:-Automator: Bump upstream}"
+}
+
+function fetch_gh_data() {
+  local raw
+  raw=$(curl -sSfLH "Authorization: token ${TOKEN}" https://api.github.com/user)
+
+  GH_USER=$(echo "${raw}" | jq --raw-output .login)
+  GH_EMAIL=$(echo "${raw}" | jq --raw-output .email)
+
+  if [ -z "${GH_USER}" ] || [ -z "${GH_EMAIL}" ]; then
+    error "Error fetching bot's data from GitHub"
+  fi
+
+  # Create a fork
+  curl -XPOST -sSfLH "Authorization: token ${TOKEN}" "https://api.github.com/repos/${ORG}/${REPO}/forks" > /dev/null
```

Note discussed offline: this will do two things: 
* ignore errors when a fork already exists
* ignore errors when GitHub has outages.

---

In the second case, curl will return an error code > 0 (which is also true when server replies with code > 400) which means that line will fail (because we `set -e`)